freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -181,4 +192,30 @@ private boolean isDuplicateEntryException(FreeIpaClientException e) {
                 .filter(c -> c == DUPLICATE_ENTRY_ERROR_CODE)
                 .isPresent();
     }
+
+    SecretResponse getSecretResponseForPrincipal(String accountId, String principal) {
+        try {
+            String path = String.format("%s/%s/%s/%s-%s", accountId,
+                    "ServiceKeytab", "serviceprincipal",
```

Comment:
Shouldn't `"serviceprincipal"` be `principal`? That way the clean up can remove it by the principal name. Or do we need to have the path contain something which is specific to the env/cluster (I think thinking of clean up in cases where this succeeds but the client gets an error back and so the client doesn't know that the cleanup methods need to be called but they do know that the cluster needs to be deleted or something along those lines). We can probably handle this better with CDPSDX-515.

(Same for `getSecretResponseForKeytab()`)

Reply:
Once we have more understanding of the cases where cleanup has to be performed we can make changes to the way is constructed based on what we find.